File: BankingApp_WDP.API/DTOs/Accounts/UpdateAccountRequest.cs
using System.ComponentModel.DataAnnotations;

namespace BankingApp.DTOs.Accounts;

public class UpdateAccountRequest
{
    [MaxLength(3)]
    public string? Currency { get; set; }

    // CurrentAccount
    [Range(0, double.MaxValue)]
    public decimal? OverdraftLimit { get; set; }

    // SavingsAccount
    [Range(0, 1)]
    public decimal? InterestRate { get; set; }

    // BusinessAccount
    [MaxLength(200)]
    public string? CompanyName { get; set; }
}

File: BankingApp_WDP.APP/app/src/main/java/com/example/bankingapp/data/model/account/UpdateAccountRequest.kt
package com.example.bankingapp.data.model.account

import com.google.gson.annotations.SerializedName

data class UpdateAccountRequest(
    @SerializedName("currency")       val currency: String?  = null,
    @SerializedName("overdraftLimit") val overdraftLimit: Double? = null,
    @SerializedName("interestRate")   val interestRate: Double?   = null,
    @SerializedName("companyName")    val companyName: String?    = null
)
